### ISSUE TITLE
Discuss recent changes to 'Thanksgiving' holidays in `kr`

### ISSUE THREAD

**Issue by ppeble** [holidays/definitions#69]
We are seeing failures when attempting to generate/test after recent updates for `kr`. The following test is now failing and we are not sure if the test is wrong, if the original definition is wrong, or if there is a legitimate bug in the ruby repository.

Link: https://github.com/holidays/definitions/blob/master/kr.yaml#L158-L163

Source test:

```yaml
  - given:
      date: '2017-10-04'
      regions: ["kr"]
      options: ["informal"]
    expect:
      name: "추석"
```

Side note: this isn't a big deal, unfortunately our current setup and is a [known issue](https://github.com/holidays/definitions/issues/42). I'm working on fixing it!

But until then, I would like some help in understanding the holidays. I am specifically referring to these: https://github.com/holidays/definitions/blob/master/kr.yaml#L47-L59

Now, I'm not a native Korean speaker (obviously! 😄 ) but google translate says the following means:

* [추석 연휴](https://github.com/holidays/definitions/blob/master/kr.yaml#L48) - 'Chuseok Holidays'
[translation](https://translate.google.com/?rlz=1C5CHFA_enUS736US736&um=1&ie=UTF-8&hl=en&client=tw-ob#auto/en/%EC%B6%94%EC%84%9D%20%EC%97%B0%ED%9C%B4)
* [추석](https://github.com/holidays/definitions/blob/master/kr.yaml#L52) - 'Thanksgiving'
[translation](https://translate.google.com/?rlz=1C5CHFA_enUS736US736&um=1&ie=UTF-8&hl=en&client=tw-ob#auto/en/%EC%B6%94%EC%84%9D)
* [추석 연휴](https://github.com/holidays/definitions/blob/master/kr.yaml#L56) - 'Chuseok Holidays'
[translation](https://translate.google.com/?rlz=1C5CHFA_enUS736US736&um=1&ie=UTF-8&hl=en&client=tw-ob#auto/en/%EC%B6%94%EC%84%9D%20%EC%97%B0%ED%9C%B4)

This is confusing to me, mainly because I don't understand the difference between 'Thanksgiving' and 'Chuseok Holidays'. As far as I can tell it's just different names for the same holiday but I could easily be wrong.

In addition, I don't fully understand the 'correct' definitions here. It seems to be traditionally a multi-day holiday. So I was hoping for some help from any of our contributors that understand the details here to help @ttwo32 and I navigate this.

Here is what I would like answered:

* Can you help us understand the difference in the holiday names as they stand today?
* Can you help us understand if our current definitions are correct?
* Can you help us understand why we need to repeat 'Chuseok Holidays' on the 14th and 16th but not the 15th, as defined?

Any additional details would be much appreciated! Pinging @swcheon @jonathanpike @marocchino for help since you are the three non-maintainers that have helped us in the past. Thank you so much!

**Comment by swcheon** [comment]
Hello pebble, I'm a native Korean speaker.
First of all, Thank you for your active reaction. :)

Here are my answers to the questions above.

1. Can you help us understand the difference in the holiday names as they stand today?
> `추석` = `Chuseok` (More appropriate translation) = `Korean Thanksgiving Day` (Easy to understand in english)
But, Google translated from `추석` into `Thansgiving`. So, I suggested that It is more appropriate for `Thanksgiving` to be translated as `Chuseok` to Google Translator.

2. Can you help us understand if our current definitions are correct?
> https://github.com/holidays/definitions/blob/master/kr.yaml#L77
I think `30th day of Dec.` should be changed to the day before `설날`. (설날: New Year's Day, https://github.com/holidays/definitions/blob/master/kr.yaml#L17)

On the lunar calendar, 2015-12-29 is the last day of year.
On the lunar calendar, 2016-12-30 is the last day of year.
The last day of last month in lunar calendar shifting from year to year.

And If it is now,
- Now (Wrong)
설날 연휴 (The day before New Year's Day): `lunar_to_solar(2017, 12, 30, :kr)`
설날 (on the Day of New Year's Day): `lunar_to_solar(2017, 1, 1, :kr)`
설날 연휴 (The day after New Year's Day): `lunar_to_solar(2017, 1, 2, :kr)`

- Correct
설날 연휴 (The day before New Year's Day): `lunar_to_solar(2016, 12, 30, :kr)`
설날 (on the Day of New Year's Day): `lunar_to_solar(2017, 1, 1, :kr)`
설날 연휴 (The day after New Year's Day): `lunar_to_solar(2017, 1, 2, :kr)`

I hope other native Korean speakers except me will make a confirmation.

3. Can you help us understand why we need to repeat 'Chuseok Holidays' on the 14th and 16th but not the 15th, as defined?
> As the day before and the day after are also part of the holiday.
To be more specific,
14th day of August in lunar calendar: The day before Chuseok
15th day of August in lunar calendar: on the Day of Chuseok
16th day of August in lunar calendar: The day after Chuseok

`연휴` is a word collectively referring to these words(The day before Chuseok, The day after Chuseok).
And, I think that would be better to change descriptions more specific.

Date / Before / After
------------------
The day before New Year's Day / 설날 연휴 / 설날 전날
2nd day of First month in lunar calendar / 설날 연휴 / 설날 다음날
14th day of August in lunar calendar / 추석 연휴 / 추석 전날
16th day of August in lunar calendar/ 추석 연휴 / 추석 다음날

I am always thankful for your efforts and contributions. :)

**Comment by ppeble** [comment]
@swcheon Thank you so much! I'm sorry that I didn't get back to this right away, just more baby stuff came up and I ran out of free time. 😄 

I will take a look at this in the coming days and formulate a plan. Thank you again!

**Comment by EunJung-Seo** [comment]
@ppeble I agree with your suggestions. `전날` and `다음날` are more correct words. But I do not use that word in the real world. I think `추석 연휴` is not so bad. How about using `추석 연휴` as it is?

**Comment by swcheon** [comment]
@EunJung-Seo I think that is also good.

**Comment by marocchino** [comment]
> Can you help us understand if our current definitions are correct?

No. After 2013, 설날 연휴 and 추석 연휴 is applied Substitute holiday.

as described in `관공서의 공휴일에 관한 규정 일부개정령안`:
If the holiday(or Sunday) falls on a New Year's Day or a Chuseok holiday, the first business day following that day shall be a holiday. If the Children's Day falls on a Saturday or another holiday (Sunday, Buddha's birthday, etc.), the first business day following that day shall be a holiday.


ref here:  <https://ko.wikipedia.org/wiki/%EB%8C%80%EC%B2%B4%ED%9C%B4%EC%9D%BC%EC%A0%9C%EB%8F%84>

**Comment by marocchino** [comment]
@swcheon Is `lunar_to_solar(2017, 12, 30, :kr)` wrong?
IMO, Both of `lunar_to_solar(2017, 12, 30, :kr)` and `lunar_to_solar(2016, 12, 30, :kr)` will be 설날 연휴.
